### PR TITLE
Move code to yast-caasp

### PR DESCRIPTION
**Pull request by imobachgs**
This PR moves some CaaSP specific code from yast-installation.

I wonder if we should use the `SLE-12-SP2-CASP` from the very beginning. In that case, it's just a matter of merging #2 and diverge them (rebasing also this PR).

### CODE REVIEW

**Comment by lslezak** [comment]
Use the `master` branch here, we use the same approach for the https://github.com/yast/skelcd-control-CASP package which is also CaaSP specific.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+*.pot
+package/*.bz2
+/coverage/
+/doc/
+/nbproject/
```
Comment (lslezak) [inline .gitignore]: That's for the NetBeans IDE, I do not use it anymore, you can drop it. :wink:

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+# TODO: this need some non-trivial refactoring...
+Metrics/PerceivedComplexity:
+  Max: 25
+
+# FIXME: It seems the cop does not handle this complex case correctly,
+# disable it for now on that file.
+Lint/ShadowedException:
+  Exclude:
+    - 'src/lib/registration/connect_helpers.rb'
```
Comment (lslezak) [inline .rubocop.yml]: This should be removed...

Comment (lslezak) [reply .rubocop.yml]: In general whole file should be cleaned up, I guess many rules are not needed. Just remove everything except the `inherit_from` option, then run Rubocop and see what fails. Then either fix that or disable.

File: MAINTAINER [hunk 3]
```diff
@@ -0,0 +1 @@
+Deprecated file. Use `osc maintainer yast2-registration` instead.
```
Comment (lslezak) [inline MAINTAINER]: Wrong package, and as it is deprecated I'd not add the file here at all.

File: package/yast2-caasp.spec [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+# case the license is the MIT License). An "Open Source License" is a
+# license that conforms to the Open Source Definition (Version 1.9)
+# published by the Open Source Initiative.
+
+# Please submit bugfixes or comments via http://bugs.opensuse.org/
+#
+
+
+Name:           yast2-caasp
+Version:        1.0.0
```
Comment (lslezak) [inline package/yast2-caasp.spec:20]: Um, I'm not sure about the versioning, maybe we should start with `3.1.0` to be consistent with the other YaST packages :question: :question: 

But 1.0.0 is fine as well... Opinions?

File: package/yast2-caasp.spec [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+# SystemRoleHandlersRunner
+BuildRequires:  yast2-installation >= 3.1.217.30
+BuildRequires:  yast2
+BuildRequires:  yast2-devtools >= 3.1.39
+BuildRequires:  rubygem(yast-rake) >= 0.2.13
+BuildRequires:  rubygem(rspec)
+
+BuildArch:      noarch
+
+# FIXME: it seems can we cannot move it to macros.yast, the yast-rake-ci is not
```
Comment (lslezak) [inline package/yast2-caasp.spec]: This is just an experimental code for building pull requests in Jenkins. As we have switched to Docker based Travis builds we do not need this anymore. Please remove this.

**Review by lslezak — CHANGES_REQUESTED** [review]

File: package/yast2-caasp.spec (hunk 4 above)
Comment (kobliha) [reply package/yast2-caasp.spec:20]: CaaSP is 1.0

but honestly, it doesn't matter

Comment (lslezak) [reply package/yast2-caasp.spec:20]: OK, so let stick with `1.0.0`. (We can always upgrade to 3.1.x if needed.)

**Review by lslezak — APPROVED** [review]

**Comment by imobachgs** [comment]
Thanks!

**Pull request merged**